docs/src/main/asciidoc/grpc-kubernetes.adoc

```diff
@@ -0,0 +1,81 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Deploying your gRPC Service in Kubernetes
+include::_attributes.adoc[]
+:categories: serialization
+:summary: This guide explains how to deploy your gRPC services in Quarkus to Kubernetes.
+
+This page explains how to deploy your gRPC service in Quarkus in Kubernetes. 
+We'll continue with the example from xref:grpc-getting-started.adoc[the Getting Started gRPC guide]. 
+
+== Configuring your project to use the Quarkus Kubernetes extension
+
+Edit the `pom.xml` file to add the Quarkus Kubernetes extension dependency (just under `<dependencies>`):
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-kubernetes</artifactId>
+</dependency>
+----
+
+Next, we want to expose our application using the Kubernetes Ingress resource:
+
+[source,properties]
+----
+quarkus.kubernetes.ingress.expose=true
+----
+
+By default, the Quarkus Kubernetes will expose the HTTP port with name `http` which is bound to the Quarkus HTTP server, not the gRPC service. To expose the gRPC server instead, set the `quarkus.kubernetes.ingress.target-port=grpc` property in your application.properties:
+
```

Comment:
How does that work with the new server where both http and gRPC are served on the same port?

Reply:
Can you share more details about this? But I guess that if both HTTP and gRPC are handing by the same port, it will work fine using the default `http` port name. 
Though, if the port name `grpc` won't be generated any longer, we might need to revisit this statement.

Reply:
With the old gRPC server, http and gRPC are served on different ports.
With the new gRPC server, HTTP and gRPC are served on the same port (which is very likely going to be https). 

Both mode are supported at the moment.

Reply:
@alesj do you think we could add tests for both cases?

Reply:
I took a deeper look into this that was implemented as part of https://github.com/quarkusio/quarkus/pull/28654.
So, by default, it creates two servers: one for HTTP and another one for gRPC. 
If `quarkus.grpc.server.use-separate-server` is `false`, then the gRPC port won't be listening. 

Note that the `grpc` port is still bound to the generated Deployment/Service resources even though `quarkus.grpc.server.use-separate-server` is false. Because `quarkus.grpc.server.use-separate-server` is a runtime property, this might make sense. So, ultimately the user would need to select the target port (either `http` or `grpc`) of the generated resource. 

We could again follow a best-effort approach, so if users use `quarkus.grpc.server.use-separate-server=false` at build time, we could remove the binding of the `grpc` port in the generated resources and use `http` by default as the target port. Wdyt @cescoffier ?

Reply:
+1 or having an explicit property on the kubernetes side.